lib/datasets/estatjp.rb

```diff
@@ -0,0 +1,302 @@
+        params['cdCat01'] = cat.join(',') if cat.instance_of?(Array)
+        # cdTime: ["1981100000", "1982100000" ,"1984100000"].join(","), # 時間軸事項
+        params['cdTime'] = time.join(',') if time.instance_of?(Array)
+
+        URI.parse("#{base_url}?#{URI.encode_www_form(params)}")
+      end
+
+      def self.extract_def(data, id)
+        rec = data['GET_STATS_DATA']['STATISTICAL_DATA']\
+        ['CLASS_INF']['CLASS_OBJ']
+        rec.select { |x| x['@id'] == id }
+      end
+
+      def self.index_def(data_def)
+        unless data_def.first['CLASS'].instance_of?(Array)
+          # convert to array when number of element is 1
+          data_def.first['CLASS'] = [data_def.first['CLASS']]
+        end
+        Hash[*data_def.first['CLASS'].map { |x| [x['@code'], x] }.flatten]
+      end
+
+      def self.get_values(data)
+        data['GET_STATS_DATA']['STATISTICAL_DATA']['DATA_INF']['VALUE']
+      end
+
+      #
+      # generate accessor instance for e-Stat API.
+      # for detail spec : https://www.e-stat.go.jp/api/api-info/e-stat-manual
+      # @param [String] api_version API Version (defaults to `'2.1'`)
+      # @param [String] stats_data_id 統計表ID
+      # @param [Array<String>] cat 分類事項 (省略時はすべて取得)
+      # @param [Array<String>] area 地域事項 (省略時はすべて取得)
+      # @param [Array<String>] time 時間軸事項 (省略時はすべて取得)
+      # @param [Array<Number>] skip_level 省略する階層レベル (defaults to `[1]`)
+      # @param [String] hierarchy_selection 階層関係の上位・末端のフィルタリングをする ('child', 'parent', 'both') (例: 札幌市○○区 -> 'child':札幌市○○区のみ; 'parent':札幌市のみ; 'both':フィルタリングしない)
+      # @param [Boolean] skip_nil_column 1行でも欠損がある列をスキップする
+      # @param [Boolean] skip_nil_row 1列でも欠損がある行をスキップする
+      # @example
+      #   estat = Datasets::Estatjp::JSONAPI.new(
+      #     "0000020201", # Ａ　人口・世帯
+      #     cat: ["A1101"], # A1101_人口総数
+      #     area: ["01105", "01106"], # "北海道 札幌市 豊平区", "北海道 札幌市 南区"
+      #     time: ["1981100000", "1982100000"],
+      #     hierarchy_selection: 'child' , # 例: 札幌市○○区があるときは札幌市をスキップ
+      #     skip_child_area: true, # 例: 札幌市○○区をスキップして札幌市を残す TODO skip_(parent|child) を統合する
+      #     skip_nil_column: true, #  1行でも欠損があったら列をスキップする
+      #     skip_nil_row: false, # 1列でも欠損があったら行をスキップする
+      #   )
+      #
+      def initialize(stats_data_id,
+                     api_version: '2.1',
+                     area: nil, cat: nil, time: nil,
+                     skip_level: [1],
+                     hierarchy_selection: 'child',
+                     skip_nil_column: true,
+                     skip_nil_row: false,
+                     time_range: nil)
+        @app_id = fetch_appid
+        if @app_id.nil? || @app_id.empty?
+          raise ArgumentError, 'Please set app_id via `Datasets::Estatjp.configure` method or environment var `ESTATJP_APPID`'
+        end
+
+        super()
+
+        @base_url = "http://api.e-stat.go.jp/rest/#{api_version}/app/json/getStatsData"
```

Comment:
How about using `htttps`?

Reply:
Yes, they support `https` so I'll fix this.